src/main/java/com/thealgorithms/dynamicprogramming/LongestArithmeticSubsequence.java

```diff
@@ -0,0 +1,45 @@
+package com.thealgorithms.dynamicprogramming;
+
+import java.util.HashMap;
+
+final class LongestArithmeticSubsequence {
+
+    private LongestArithmeticSubsequence() {
+    }
+
+    /**
+     * Returns the length of the longest arithmetic subsequence in the given array.
+     *
+     * A sequence seq is arithmetic if seq[i + 1] - seq[i] are all the same value
+     * (for 0 <= i < seq.length - 1).
+     *
+     * @param nums the input array of integers
+     * @return the length of the longest arithmetic subsequence
+     */
+    public static int getLongestArithmeticSubsequenceLength(int[] nums) {
+        // If the array is empty or has only one element, return its length.
+        if (nums == null || nums.length == 0) {
+            return 0;
+        }
+
+        if (nums.length == 1) {
+            return 1;
+        }
+
+        int n = nums.length;
```

Comment:
```suggestion
        final int n = nums.length;
```
I am not sure if creating this variable is needed - there is not overhead of using `nums.length` directly.